lib/cmsis_rtos_v1/cmsis_mailq.c

```diff
@@ -110,11 +110,8 @@ osStatus osMailPut(osMailQId queue_id, void *mail)
 	mmsg.rx_source_thread = K_ANY;
 	mmsg.tx_target_thread = K_ANY;
 
```

Comment:
> osErrorValue: mail was previously not allocated as memory slot.

I think in `osMailPut` function, this check should be added i.e. `if (mail == NULL) {return osErrorValue;}` before assigning it to the mail box item in the queue in line 109; so if it is NULL or not allocated from `osMailAlloc` in the case:

> A NULL pointer is returned when no memory block can be obtained

this function does not continue to put an empty mail in the queue.

Reply:
Addressed. Thanks for pointing this out.